src/url.rs

```diff
@@ -0,0 +1,389 @@
+                                        )),
+                                    }
+                                }
+                                Err(e) => Err(format!(
+                                    "{} - Failed dumping graph as turtle: {}",
+                                    fdk_id, e
+                                )),
+                            }
+                        }
+                        Err(e) => Err(format!(
+                            "{} - Failed creating new metrics store: {}",
+                            fdk_id, e
+                        )),
+                    }
+                }
+                None => Err(format!("{} - Dataset node not found in graph", fdk_id)),
+            }
+        }
+        Err(e) => Err(format!("{} - Failed to parse graph: {}", fdk_id, e)),
+    }
+}
+
+fn check_urls(
+    fdk_id: String,
+    dataset_node: NamedNodeRef,
+    store: &Store,
+) -> Result<Store, StorageError> {
+    // Make MQA metrics model (DQV)
+    let metrics_store = create_metrics_store(dataset_node)?;
+    let distributions =
+        list_distributions(dataset_node, store).collect::<Result<Vec<Quad>, _>>()?;
+
+    for dist in distributions {
+        metrics_store.insert(
+            Quad::new(
+                dist.subject.clone(),
+                dist.predicate.clone(),
+                dist.object.clone(),
+                GraphName::DefaultGraph,
+            )
+            .as_ref(),
+        )?;
+
+        let dist_node = match dist.object {
+            Term::NamedNode(n) => Some(NamedOrBlankNode::NamedNode(n)),
+            Term::BlankNode(n) => Some(NamedOrBlankNode::BlankNode(n)),
+            _ => None,
+        }
+        .unwrap();
+
+        info!("{} - Extracting urls from distribution", fdk_id);
+        let urls = extract_urls_from_distribution(dist_node.as_ref(), &store);
+        info!("{} - Number of urls found {}", fdk_id, urls.len());
+
+        for url in urls {
+            let result = check_url(
+                url.get("method").unwrap().to_string(),
+                if url.contains_key("access_url") {
+                    url.get("access_url").unwrap().to_string()
+                } else {
+                    url.get("download_url").unwrap().to_string()
+                },
+                if url.contains_key("access_url") {
+                    UrlType::AccessUrl
+                } else {
+                    UrlType::DownloadUrl
+                },
+            );
+
+            info!("{}", result.note);
+
+            let url_node: NamedNodeRef<'_> = NamedNodeRef::new_unchecked(result.url.as_str());
+
+            match result.url_type {
+                UrlType::AccessUrl => add_access_url_status_metric(
+                    dist_node.as_ref(),
+                    url_node,
+                    result.status,
+                    &metrics_store,
+                )?,
+                UrlType::DownloadUrl => add_download_url_status_metric(
```

Comment:
`add_access_url_status_metric` og `add_download_url_status_metric` er så like, og brukes så likt, at jeg ville vurdert å ha én funksjone for begge, som tar inn `UrlType`, og håndterer forskjellen inni funksjonen.